src/plugins/stake-plugins/thorchainSavers/tcSaversPlugin.tsx

```diff
@@ -1087,6 +1140,220 @@ const unstakeRequestInner = async (opts: EdgeGuiPluginOptions, request: ChangeQu
+      pluginId: stakeProviderInfo.pluginId,
+      stakeAssets: [
+        {
+          pluginId,
+          tokenId,
+          nativeAmount
+        }
+      ]
+    }
+  }
+
+  if (isEvm) {
+    // For mainnet coins of EVM chains, use gasLimit override since makeSpend doesn't
+    // know how to estimate an ETH spend with extra data
+    spendInfo.networkFeeOption = 'custom'
+    spendInfo.customNetworkFee = {
+      ...spendInfo.customNetworkFee,
+      gasLimit: EVM_SEND_GAS
+    }
+  }
+
+  let networkFee = '0'
+  let needsFundingPrimary = false
+  if (lt(addressBalance, nativeAmount)) {
+    // Easy check to see if primary address doesn't have enough funds
+    if (isEvm) {
+      // EVM chains only have one address, so if there aren't enough funds in
+      // the primary address then we don't have enough funds at all
+      throw new InsufficientFundsError({ tokenId: null })
+    }
+    needsFundingPrimary = true
+  } else {
+    try {
+      const estimateTx = await wallet.makeSpend(spendInfo)
+      networkFee = estimateTx.parentNetworkFee ?? estimateTx.networkFee
+    } catch (e: unknown) {
+      if (!isEvm && asMaybeInsufficientFundsError(e) != null) {
+        needsFundingPrimary = true
+      } else {
+        throw e
+      }
+    }
+  }
+
+  let fundingSpendInfo: EdgeSpendInfo | undefined
+  if (needsFundingPrimary) {
+    // Estimate the total cost to create the two transactions
+    // 1. Fund the primary address with the requestedAmount + fees for tx #2
+    // 2. Send the requested amount to the pool address
+    fundingSpendInfo = {
+      tokenId: null,
+      spendTargets: [
+        {
+          publicAddress: primaryAddress,
+          nativeAmount: nativeAmount
+        }
+      ],
+      assetAction: { assetActionType: 'stakeNetworkFee' },
+      savedAction: {
+        actionType: 'stake',
+        pluginId: stakeProviderInfo.pluginId,
+        stakeAssets: [
+          {
+            pluginId,
+            tokenId,
+            nativeAmount
+          }
+        ]
+      },
+      otherParams: { forceChangeAddress, enableRbf: false }
+    }
+
+    const estimateTx = await wallet.makeSpend(fundingSpendInfo)
+    networkFee = estimateTx.networkFee
+
+    // The actual funding transaction will need to fund enought for the staking amount
+    // plus the fees for the staking transaction
+    fundingSpendInfo.spendTargets[0].nativeAmount = add(networkFee, nativeAmount)
+
+    const remainingBalance = sub(sub(walletBalance, mul(networkFee, '2')), nativeAmount)
+    if (lt(remainingBalance, '0')) {
+      throw new InsufficientFundsError({ tokenId: null })
+    }
+  }
+
+  const fee = needsFundingPrimary ? mul(networkFee, '2') : networkFee
+
+  const claimableTcy = await fetchClaimableTcy(opts, primaryAddress)
+
+  let quoteInfo: QuoteInfo | undefined
+  const allocations: QuoteAllocation[] = [
+    {
+      allocationType: 'stake',
+      pluginId,
+      currencyCode,
+      nativeAmount,
+      lockInputs: true
+    },
```

Comment:
Seems like this allocation shouldn't be here. You are specifying a `claim` so the small dust you are sending should be specified in the 'networkFee' and what you are claiming is specified below.

Reply:
Please bounce this off Sam